pkg/extended_registry/extended_registry.go

```diff
@@ -21,111 +19,86 @@ type Config struct {
 	Password string `json:"password"`
 }
 
-type ExtendedClient struct {
-	cfg       *Config
-	logger    *zerolog.Logger
-	transport *http.Transport
+type ExtendedClient interface {
+	ListOrgs() ([]string, error)
+	ListRepos(org string) ([]*api.PolicyImage, error)
+	ListTags(org, repo string, page *api.PaginationRequest) ([]*api.RegistryRepoTag, *api.PaginationResponse, error)
+	GetTag(org, repo, tag string) (*api.RegistryRepoTag, error)
+	SetVisibility(org, repo string, public bool) error
+	RemoveImage(org, repo, tag string) error
+	IsValidTag(org, repo, tag string) (bool, error)
 }
 
-type PolicyImage struct {
-	Name   string `json:"name"`
-	Public bool   `json:"public"`
+type xClient struct {
+	cfg    *Config
+	logger *zerolog.Logger
+	client *http.Client
 }
 
-func NewExtendedClient(logger *zerolog.Logger, cfg *Config, transport *http.Transport) *ExtendedClient {
-	return &ExtendedClient{
-		cfg:       cfg,
-		logger:    logger,
-		transport: transport,
+func newExtendedClient(logger *zerolog.Logger, cfg *Config, client *http.Client) *xClient {
+	return &xClient{
+		cfg:    cfg,
+		logger: logger,
+		client: client,
 	}
 }
 
-func (c *ExtendedClient) Supported() bool {
-	address, err := c.extendedAPIAddress()
-	if err != nil {
-		c.logger.Trace().Err(err).Msg("failed to get extended API address")
-		return false
+func GetExtendedClient(server string, logger *zerolog.Logger, cfg *Config, transport *http.Transport) (ExtendedClient, error) {
+	httpClient := http.Client{}
+	httpClient.Transport = transport
+	switch {
+	case server == "opcr.io", strings.Contains(server, "aserto"):
```

Comment:
I think we need to be careful making assumptions on the clint based on the server URL. 
For example:
registry.prod.opcr.io points to opcr.io